---
title: "scRNA analysis"
date: "`r Sys.Date()`"
output: 
  html_document:
    toc: true # table of content true
    toc_depth: 1
    toc_float:
      collapsed: true
      smooth_scroll: true
    number_sections: false  ## if you want number sections at each table header
    theme: united  # many options for theme, this one is my favorite.
    highlight: tango  # specifies the syntax highlighting style
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning=FALSE, message=FALSE)

library(stringr)
library(dplyr)
library(Seurat)
library(patchwork)
library(ggplot2)
library(openxlsx)
library(harmony)
library(SingleR)

# Directories
proj_path <- "/.."
data_path <-  file.path(proj_path, "../")
sc_sample_paths <- paste0(data_path, rep(c("aging_F7", "aging_F15", "aging_M4", "aging_M11")))
count_paths <- file.path(sc_sample_paths, "outs/filtered_feature_bc_matrix")


result_path <-  "../result"
experiment <- "Aging"
experiment_path <- file.path(result_path, experiment)
dir.create(experiment_path, recursive=TRUE)

out_path <- experiment_path

## settings
sample_names <- c("F7_KO", "F15_WT", "M4_KO", "M11_WT")

#mtPattern: a regex pattern to match against features to look for mitochondrial DNA; may need to adjust for 
#sample species---"^MT-" works for human samples and "^mt-" for mice
mtPattern <- "^mt-"   
doCite <- FALSE

```

```{r load}

seurats <- list()

for (s1 in 1:length(sample_names)){
  input.data <- Read10X(count_paths[s1])
  seurat_obj <- CreateSeuratObject(counts = input.data, project = sample_names[s1])
  if(doCite){
    adt.data <-  Read10X(adt_count_paths[s1], gene.column = 1)
    colnames(adt.data) <- paste0(colnames(adt.data), "-1")
    adt_assay <- CreateAssayObject(counts = adt.data)
    seurat_obj[["ADT"]] <- adt_assay
  }
  seurats[[sample_names[s1]]] <- seurat_obj
}
```

```{r test}
pbmc <- seurat_obj
pbmc[["percent.mt"]] <- PercentageFeatureSet(pbmc, pattern = "^MT-")
VlnPlot(pbmc, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

```

# Quality Check

```{r QC}

for (s1 in sample_names){
  # Add fields to seurat object's metadata
  seurat_obj <- seurats[[s1]]
  
  seurat_obj[["percent.mt"]] <- PercentageFeatureSet(seurat_obj, pattern = mtPattern)
  seurat_obj$log10GenesPerUMI <- log10(seurat_obj$nFeature_RNA) / log10(seurat_obj$nCount_RNA)
  
  seurats[[s1]] <- seurat_obj
}
```

Looping over QC plots
1:09
```{r qcPlots, include=TRUE, fig.width=15}
# Violin Plot
violin_list <- list()
dotplot_list <- list()
metadata_list <- list()
for(v1 in names(seurats)){
  cat("### QC Plots ", v1, "\n")
  vPlot <- VlnPlot(seurats[[v1]], features=c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol=3,
                     log = TRUE) + NoLegend()
  ggsave(filename = file.path(out_path, paste0("qc_violin_plot_", v1,".png")))
  violin_list[[v1]] <- vPlot
  metadata <- seurats[[v1]]@meta.data
  metadata_list[[v1]] <- metadata
  d1 <- metadata %>%
    ggplot(aes(x=nCount_RNA, y=nFeature_RNA, color=percent.mt)) +
    geom_point() + ggtitle("RNA1") +
    scale_colour_gradient(low = "gray90", high = "black") +
    stat_smooth(method=lm) +
    scale_x_log10() +
    scale_y_log10() +
    theme_classic() +
    geom_vline(xintercept = 500) +
    geom_hline(yintercept = 250)
  ggsave(filename = file.path(out_path, paste0("genes_UMI_corr_plot_", v1,".png")))
  dotplot_list[[v1]] <- d1
  cat("\n")
  print(vPlot | d1)
  cat("\n")
}
```


```{r merge_objects}

seurat_obj <- merge(x = seurats[[1]], y = seurats[2:length(seurats)], add.cell.id = sample_names)

# Add sample names to metadata:
# Pull out seurat_obj's metadata as a dataframe
metadata <- seurat_obj@meta.data
# Add cell names as column
metadata$"cells" <- rownames(metadata)
# Add sample names to metadata
metadata$sample <- NA
for (s in sample_names){
  nameStr <- paste0("^", s)
  metadata$sample[which(str_detect(metadata$cells, nameStr))] <- s
}
# Add metadata back into Seurat object
seurat_obj@meta.data <- metadata
```

## Pre-filtering cellcount

```{r orig_counts}
orig_idents <- table(seurat_obj$orig.ident)
knitr::kable(orig_idents, caption = "Pre-Filtering Cell Counts")
```

# Filtering

```{r filtering}
nCount_cutoff <- 10
nFeature_cutoff <- 100
genesPerUMI_cutoff <- 0.5
pctMt_cutoff <- 10
cellsPerGene_cutoff <- 10

# Filter out low quality reads using selected thresholds - these will change with experiment
filtered_seurat <- subset(x = seurat_obj, 
                          subset= (nCount_RNA >= nCount_cutoff) & 
                            (nFeature_RNA >= nFeature_cutoff) & 
                            (log10GenesPerUMI > genesPerUMI_cutoff) & 
                            (percent.mt < pctMt_cutoff))

# Filter out features with too few cells per gene
# Extract counts

filtered_seurat <- JoinLayers(filtered_seurat)
counts <- GetAssayData(object = filtered_seurat, slot = "counts")
# Output a logical vector for every gene on whether more than zero counts per cell
nonzero <- counts > 0
# Sums all TRUE values and returns TRUE if more than <cutoff #> TRUE values per gene
keep_genes <- Matrix::rowSums(nonzero) >= cellsPerGene_cutoff
# Only keeping those genes expressed in more than <cutoff #> cells
filtered_counts <- counts[keep_genes, ]

# Reassign to filtered Seurat object
filtered_seurat <- CreateSeuratObject(filtered_counts, meta.data = filtered_seurat@meta.data)

filtered_table <- table(filtered_seurat$orig.ident)
```

## Cutoffs

nCount minimum: `r nCount_cutoff`

nFeature minimum: `r nFeature_cutoff`

log10 genes per UMI minimum: `r genesPerUMI_cutoff`

Percent MT DNA maximum: `r pctMt_cutoff`

Minimum cells per gene: `r cellsPerGene_cutoff`

## Post-filtering cell counts

```{r filtering2}
#saveRDS(filtered_seurat, file = paste0(out_path, "filtered_seurat_object.rds"))

knitr::kable(filtered_table, caption = "Filtered Cell Counts")
```

``` {r loading filtered seurat object}
#filtered_seurat <- readRDS(paste0(out_path, "Agingfiltered_seurat_object.rds"))
```
## Post-filtering Quality

```{r filtering3}
Idents(filtered_seurat) <- filtered_seurat$orig.ident
VlnPlot(filtered_seurat, features=c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol=3,
                     log = TRUE, pt.size = 0) + NoLegend()
ggsave(filename = file.path(out_path, "filtered_violin_plot_all.png"))


metadata <- filtered_seurat@meta.data
metadata %>% 
  ggplot(aes(x=nCount_RNA, y=nFeature_RNA, color=percent.mt)) + 
  geom_point() + 
  scale_colour_gradient(low = "gray90", high = "black") +
  stat_smooth(method=lm) +
  scale_x_log10() + 
  scale_y_log10() + 
  theme_classic() +
  geom_vline(xintercept = 500) +
  geom_hline(yintercept = 250) 
ggsave(filename = file.path(out_path, "filtered_genes_UMI_corr_plot.png"))
```

# Normalization, PCA

## Elbow Plot (PCA loadings)

```{r normalization}
# Normalize RNA data with log normalization
norm_seurat <- NormalizeData(filtered_seurat)

# Find and scale variable features
norm_seurat <- FindVariableFeatures(norm_seurat, selection.method = "mean.var.plot")
norm_seurat <- ScaleData(norm_seurat, features = VariableFeatures(norm_seurat))

# Run PCA
pca_seurat <- RunPCA(norm_seurat)

ElbowPlot(pca_seurat, ndims=20, reduction = "pca")
```

```{r integration}
# he RunHarmony function in Seurat applies the Harmony algorithm to correct for batch effects in integrated analysis of scRNA-seq data
integ_seurat <- RunHarmony(pca_seurat, group.by.vars = "sample")
## Integration was not needed, so decided to use merged data for further analysis
#integ_seurat <- pca_seurat
```


# Integration with Harmony and Clustering

```{r clustering1}
# Set number of dimensions & resolution values to be used in clustering
nDims <- 18
cluster_resolution <- 0.8

```
Number of dimensions: `r nDims`
Cluster resolution: `r cluster_resolution`
# The function FindVariableFeatures identifies the most highly variable genes (default 2000 genes) by fitting a line to the relationship of log(variance) and log(mean) using loess smoothing, uses this information to # # standardize the data, then calculates the variance of the standardized data. This helps avoid selecting genes that only appear variable due to their expression level.
```{r Find Variable Features}

integ_seurat <- FindVariableFeatures(
  object = integ_seurat,
  selection.method = "vst")
top10 <- head(VariableFeatures(integ_seurat), 10)

length(VariableFeatures(integ_seurat))

vfp1 <- VariableFeaturePlot(integ_seurat)
vfp1 <- LabelPoints(plot = vfp1, points = top10, repel = TRUE)

```


```{r clustering2}
cluster_seurat <- FindNeighbors(integ_seurat, reduction="pca", dims = 1:nDims)
cluster_seurat <- FindClusters(cluster_seurat, resolution = cluster_resolution, verbose = FALSE)
cluster_seurat <- RunUMAP(cluster_seurat, dims = 1:nDims)

DimPlot(cluster_seurat, label = TRUE)
ggplot2::ggsave(file.path(out_path, "clusters_all.png"))


```


``` {r}

#saveRDS(cluster_seurat, file = paste0(out_path, "cluster_seurat.rds"))
cluster_seurat <- readRDS("/ix/djishnu/Zarifeh/ML_MM/Aditi/result/Aging/Agingcluster_seurat.rds")

```
# Finding differentially expressed features (cluster biomarkers)

```{r degs}

aging.markers <- FindAllMarkers(cluster_seurat, only.pos = TRUE)
aging.markers %>%
    group_by(cluster) %>%
    dplyr::filter(avg_log2FC > 1)
write.xlsx(x=aging.markers, file=file.path(out_path, "all_markers.xlsx"))

```

``` {r up and downregulated}

aging.markers_updown <- FindAllMarkers(cluster_seurat, only.pos = FALSE)
aging.markers_updown %>%
    group_by(cluster) %>%
    dplyr::filter(avg_log2FC > 1)
write.xlsx(x=aging.markers_updown, file=file.path(out_path, "all_markers_updown.xlsx"))

```
``` {r cluster 0}
cluster0.markers <- FindMarkers(cluster_seurat, ident.1 = 0, logfc.threshold = 1, test.use = "roc", only.pos = FALSE)
head(cluster0.markers)
```
``` {r cluster 1}
cluster1.markers <- FindMarkers(cluster_seurat, ident.1 = 1, logfc.threshold = 1, test.use = "roc", only.pos = FALSE)
head(cluster1.markers)
```
``` {r cluster 2}
cluster2.markers <- FindMarkers(cluster_seurat, ident.1 = 2, logfc.threshold = 1, test.use = "roc", only.pos = FALSE)
head(cluster2.markers)
```

``` {r cluster 3}
cluster3.markers <- FindMarkers(cluster_seurat, ident.1 = 3, logfc.threshold = 1, test.use = "roc", only.pos = FALSE)
head(cluster3.markers)
```

``` {r cluster 4}
cluster4.markers <- FindMarkers(cluster_seurat, ident.1 = 4, logfc.threshold = 1, test.use = "roc", only.pos = FALSE)
head(cluster4.markers)
```

``` {r Assigning cell type identity to clusters }
aging.markers <- read.xlsx("/ix/djishnu/Zarifeh/ML_MM/Aditi/result/Aging/all_markers_updown.xlsx")

aging.markers %>%
    group_by(cluster) %>%
    dplyr::filter(avg_log2FC > 1) %>%
    slice_head(n = 5) %>%
    ungroup() -> top5
DoHeatmap(cluster_seurat, features = top5$gene) + NoLegend()
ggsave("/ix/djishnu/Zarifeh/ML_MM/Aditi/result/Aging/cluster_heatmap_top5.png", device = "png", width = 20, height = 30) 

top5$gene
DotPlot(object = cluster_seurat, 
        features = unique(top5$gene),
        cluster.idents = TRUE,
        scale.by = "radius",
        cols = c("lightgrey", "darkgreen"),
        
         )+ RotatedAxis()

ggsave("../DOTPLOT.png", device = "png", width = 18, height = 15) 


```


``` {r Feature plots }
#senescence markers: 
FeaturePlot(cluster_seurat, features = c("Cdkn1a","Myh8","Col19a1","Lrrk2","Acta2"), ncol = 3)
ggplot2::ggsave(file.path(out_path, "feature_plot_all_1.png"))

```



``` {r making reference}
cell_annot <- data.frame(
  Cell_type = c(
    "MuSC", "MF-I", "MF-IIB", "MF-IIX", "MF-IIA", 
    "NMJ", "MTJ", "MF-other", "FB", "Tenocyte", 
    "mSchwann", "nmSchwann", "SMC", "Pericyte", 
    "CapEC", "ArtEC", "VenEC", "LymphEC", "M2_LYVE1", 
    "Macrophages_LAM", "Monocyte", "Neutrophil", 
    "B cell", "Adipocyte", "RBC"
  ),
  Markers = c(
    "Pax7", "Lpl, Tnnt1, Tnnc1, Tnnt3, Tnnc2", 
    "Tnnt3, Tnnc2", "Myh1, Tnnc2, Tnnt3", 
    "Lpl, Myh2, Tnnc2, Tnnt3", "Tnnc2, Tnnt3", 
    "Col22a1, Tnnc2, Tnnt3, Rgcc", 
    "Tnnc2, Tnnt3, Rgcc", "Lpl, Col1a1, Dcn, Pdgfra", 
    "Lpl, Acta2, Tnmd, Col1a1, Dcn", 
    "Mpz, Mbp, Rgcc", "Ncam1, Mbp, Rgcc", 
    "Acta2, Myh11", "Acta2, Kcnj8, Rgs5, Pecam1, Cdh5, Rgcc, Lpl", 
    "Pecam1, Cdh5, Rgcc, Lpl", "Pecam1, Cdh5, Rgcc", 
    "Pecam1, Cdh5", "Ccl21a, Mmm1, Rgs5", 
    "Lyve1, F13a1, Fcgr3, Clec10a, Cd14", 
    "Fcgr3, Spp1, F13a1, Lyve1", "Fcgr3, Cd14", 
    "S100a8, Fcgr3, Cd14", "Cd79a", 
    "Cyp2e1, Lpl, Tnnt3", "Hba-a1, Lpl, Rgcc, Cdh5, Rgcc, Pecam1"
  ),
  stringsAsFactors = FALSE
)

# Write the data frame to a CSV file
write.csv(cell_annot, file = "/ix/djishnu/Zarifeh/ML_MM/Aditi/result/cell_annot.csv", row.names = FALSE)
```



``` {r cluster annotation}

aging.markers_updown <- FindAllMarkers(cluster_seurat, only.pos = FALSE)

```

``` {r cluster annotation}
print(cell_annot)

```

``` {r}

```



### Heatmap showing top 5 DE genes in each cluster

```{r heatmap, eval=FALSE}

# Load the CSV file containing cell types and markers
cell_annot <- read.csv("/ix/djishnu/Zarifeh/ML_MM/Aditi/result/cell_annot.csv", stringsAsFactors = FALSE)

aging.markers <- read.xlsx("/ix/djishnu/Zarifeh/ML_MM/Aditi/result/Aging/all_markers_updown.xlsx")
#allmarkers %>%
 # group_by(cluster) %>%
 # top_n(n = 10, wt = avg_log2FC) -> top10
#DoHeatmap(cluster_seurat, features = top5$gene) + NoLegend()

aging.markers %>%
    group_by(cluster) %>%
    dplyr::filter(avg_log2FC > 1) %>%
    slice_head(n = 20) %>%
    ungroup() -> top20

current_markers <- unlist(strsplit(cell_annot[, "Markers"], ", "))
shared <- intersect(top20$gene, current_markers)
top20_shared <- top20[top20$gene %in% shared, ] 
## Cluster identification 
FeaturePlot(cluster_seurat, features = shared[1:4], ncol = 2)
ggsave(file.path(out_path, "markers_1.png"))
FeaturePlot(cluster_seurat, features = shared[5:8], ncol = 2)
ggsave(file.path(out_path, "markers_2.png"))
FeaturePlot(cluster_seurat, features = shared[9:12], ncol = 2)
ggsave(file.path(out_path, "markers_3.png"))
FeaturePlot(cluster_seurat, features = shared[13], ncol = 2)
ggsave(file.path(out_path, "markers_4.png"))


```
``` {r}
```

# Reclustering CD4+ Tcells

## Gene marker visualiztion



## Renaming clusters based on scores

```{r TcellsOnly_genetic_markers_rename}

new.cluster.ids <- c("0","1" , "MF-IIX","3" ,"4" ,"MF-IIX","6" ,"7" ,"MF-IIA" , "9","FB,Tenocyte","CapEC,ArtEC,VenEC","MTJ","Pericyte", "MuSC", "nmSchwann", "Neuronmuscular Junction","17")

names(new.cluster.ids) <- levels(cluster_seurat)
cluster_seurat <- RenameIdents(cluster_seurat, new.cluster.ids)
DimPlot(cluster_seurat, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
ggsave(file.path(out_path, "clusters_annot.png"))

```
``` {r more markers from the paper}
FeaturePlot(cluster_seurat, features = c("Xist", "Fhl1", "Pdk4", "Fat3", "Myh1"), ncol = 3)

```
``` {r cluster identification using all the markers from the paper}
Markers = c(
    "Pax7", "Lpl", "Tnnt1", "Tnnc1", "Tnnt3", "Tnnc2", 
    "Tnnt3", "Tnnc2", "Myh1", "Tnnc2", "Tnnt3", 
    "Lpl", "Myh2", "Tnnc2", "Tnnt3", "Tnnc2", "Tnnt3", 
    "Col22a1", "Tnnc2", "Tnnt3", "Rgcc", 
    "Tnnc2", "Tnnt3", "Rgcc", "Lpl", "Col1a1", "Dcn", "Pdgfra", 
    "Lpl", "Acta2", "Tnmd", "Col1a1", "Dcn", 
    "Mpz", "Mbp", "Rgcc", "Ncam1", "Mbp", "Rgcc", 
    "Acta2", "Myh11", "Acta2", "Kcnj8", "Rgs5", "Pecam1", "Cdh5", "Rgcc", "Lpl", 
    "Pecam1", "Cdh5", "Rgcc", "Lpl", "Pecam1", "Cdh5", "Rgcc", 
    "Pecam1", "Cdh5", "Ccl21a", "Mmm1", "Rgs5", 
    "Lyve1", "F13a1", "Fcgr3", "Clec10a", "Cd14", 
    "Fcgr3", "Spp1", "F13a1", "Lyve1", "Fcgr3", "Cd14", 
    "S100a8", "Fcgr3", "Cd14", "Cd79a", 
    "Cyp2e1", "Lpl", "Tnnt3", "Hba-a1", "Lpl", "Rgcc", "Cdh5", "Rgcc", "Pecam1"
  )
#FeaturePlot(cluster_seurat, features = Markers[1:4], ncol = 2)
VlnPlot(cluster_seurat, features = unique(Markers), assay = "RNA",
        group.by = "seurat_clusters",
        stack = TRUE, flip = TRUE
        )
ggsave(file.path(out_path, "markers_paper.png"))

```

``` {r cluster identification using all immune cell markers}
marker_genes <-
  c("Itgam", # Cd11b - Myeloid marker (mouse)
    "Adgre1", # F4/80,
    "Pax5",  # B
    "Ighd", "Cd27", # Naive (mouse) B cell markers (IgD+, CD27-)
    "Cd3d", "Cd3e", "Cd8a", # T
    "Klrb1c", "Prf1", "Klrk1", "Gzma", "Gzmb",  # NK
    "Itga2", "Ncam1",  #NK-T
    "Cd83",  # DCs
    "Cd14", "Cd68",  # Macs - note that Cd16 never comes up
    "Itgax", # DCs
    "Ly6c1",
    "Cd74", # MHC-II mouse marker (used by Renthal 2022 to identify immune cells in TG)
    "Ptgs2", "Irf5", "Nos2",  # Mouse M1 Mac Markers
    "Stat1", "Retnla",  # Mouse M1 Mac Markers (less helpful)
    "Il12a", "Il23a", "Cd163",  # M1 vs M2 (M1: IL-12 and IL23 high with CD163 neg and M2 the opposite)
    "Cd163",  # M2
    "Arg1", # M2a
    "Socs3", "Cd86", # M2b
    "Ccr2", "Slamf6",   #M2c
    "Tlr1", "Tlr8", "Scarb1", #M2c (less helpful)
    "Vegfa",    # M2d,
    "Cx3cr1"  # Tissue-res Mac
  )

VlnPlot(cluster_seurat, features = marker_genes, assay = "RNA",
        group.by = "seurat_clusters",
        stack = TRUE, flip = TRUE
        )
ggsave(file.path(out_path, "immune_markers_vlnplot.png")
       
FeaturePlot(cluster_seurat, features = "Pdk4")
```

```{r TcellsOnly_genetic_markers_rename after immune markers}
#new.cluster.ids <- c("0","1" , "2","3" ,"4" ,"MF-IIX","6" ,"7" ,"MF-IIA" , "9","FB,Tenocyte","CapEC,ArtEC,VenEC","MTJ","Pericyte", "MuSC", "nmSchwann", "nmSchwann","17")

new.cluster.ids <- c("0","1" , "MF-IIX","3" ,"4" ,"MF-IIX","6" ,"7" ,"MF-IIA" , "9","FB,Tenocyte","CapEC,ArtEC,VenEC","MTJ","Pericyte", "MuSC", "nmSchwann", "Neuronmuscular Junction","F4/80, M2")

names(new.cluster.ids) <- levels(cluster_seurat)
cluster_seurat <- RenameIdents(cluster_seurat, new.cluster.ids)
DimPlot(cluster_seurat, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
ggsave(file.path(out_path, "clusters_annot.png"))

FeaturePlot(cluster_seurat, features = "Pdk4")

```

```{r markers from https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9792217/ }


FeaturePlot(cluster_seurat, features = c("Csf1r", "Nkg7", "Cd3e", "Pdgfra", "Pecam1", "Pax7", "Cd34", "Tnnt1", "Tnn"),ncol = 3) 
ggsave(file.path(out_path, "clusters_annot_PMC9792217markers.png"))
#LRRK2, CDKN1A, MYH8, COL19A1
FeaturePlot(cluster_seurat, features = c("Kdr"),ncol = 3) 
```
```{r markers in http://cloud.capitalbiotech.com/SingleCellBase/search_result.jsp}
FeaturePlot(cluster_seurat, features = c("Ccl17", "Ccl22", "Ccr7", "Pecam1", "lgbp7", "Ctla2a", "Col4a1"),ncol = 3)

```

``` {r cluster identification using all immune cell markers}
marker_genes <-
  c("Itgam", # Cd11b - Myeloid marker (mouse)
    "Adgre1", # F4/80,
    "Pax5",  # B
    "Ighd", "Cd27", # Naive (mouse) B cell markers (IgD+, CD27-)
    "Cd3d", "Cd3e", "Cd8a", # T
    "Klrb1c", "Prf1", "Klrk1", "Gzma", "Gzmb",  # NK
    "Itga2", "Ncam1",  #NK-T
    "Cd83",  # DCs
    "Cd14", "Cd68",  # Macs - note that Cd16 never comes up
    "Itgax", # DCs
    "Ly6c1",
    "Cd74", # MHC-II mouse marker (used by Renthal 2022 to identify immune cells in TG)
    "Ptgs2", "Irf5", "Nos2",  # Mouse M1 Mac Markers
    "Stat1", "Retnla",  # Mouse M1 Mac Markers (less helpful)
    "Il12a", "Il23a", "Cd163",  # M1 vs M2 (M1: IL-12 and IL23 high with CD163 neg and M2 the opposite)
    "Cd163",  # M2
    "Arg1", # M2a
    "Socs3", "Cd86", # M2b
    "Ccr2", "Slamf6",   #M2c
    "Tlr1", "Tlr8", "Scarb1", #M2c (less helpful)
    "Vegfa",    # M2d,
    "Cx3cr1"  # Tissue-res Mac
  )

VlnPlot(cluster_seurat, features = marker_genes, assay = "RNA",
        group.by = "seurat_clusters",
        stack = TRUE, flip = TRUE
        )
ggsave(file.path(out_path, "immune_markers_vlnplot.png")
       
FeaturePlot(cluster_seurat, features = "Flnc")
```

```{r from new re Aditi has sent}


marker_genes <- c("Cd55", "Pi16", "Igfbp5", "Dpp4", "Ly6a", "Hif1a", "Lpl", "Fbln5", "Cxcl12", "Thy1", "Tnfaip6", "Cd164",
                  "Cd34", "Il33", "Hmgb2", "Cdk1", "Egfr", 
                  "Smc4", "Mcm5", "Tyms", "Rrm2", "Dio2", "Rspo3"
                 )

VlnPlot(cluster_seurat, features = marker_genes, assay = "RNA",
        group.by = "seurat_clusters",
        stack = TRUE, flip = TRUE
        )
ggsave(file.path(out_path, "vlnplot6.png"))
       

```